autoPyTorch/data/utils.py

```
@@ -0,0 +1,309 @@
+            The data to reduce precision of.
+
+    Returns:
+        Tuple[DatasetCompressionInputType, DatasetDTypeContainerType, DatasetDTypeContainerType]
+            Returns the reduced data X along with the dtypes it and the dtypes it was reduced to.
+    """
+    precision: Optional[DatasetDTypeContainerType] = None
+    if isinstance(X, np.ndarray) or issparse(X):
+        dtypes = X.dtype
+        if X.dtype not in supported_precision_reductions:
+            raise ValueError(f"X.dtype = {X.dtype} not equal to any supported"
+                             f" {supported_precision_reductions}")
+        precision = reduction_mapping[X.dtype]
+        X = X.astype(precision)
+    elif hasattr(X, 'iloc'):
+        dtypes = dict(X.dtypes)
+
+        integer_columns = []
+        float_columns = []
+
+        for col, dtype in dtypes.items():
+            if is_numeric_dtype(dtype):
+                if is_float_dtype(dtype):
+                    float_columns.append(col)
+                else:
+                    integer_columns.append(col)
+
+        if len(integer_columns) > 0:
+            X[integer_columns] = X[integer_columns].apply(lambda column: pd.to_numeric(column, downcast='integer'))
+        if len(float_columns) > 0:
+            X[float_columns] = X[float_columns].apply(lambda column: pd.to_numeric(column, downcast='float'))
+        precision = dict(X.dtypes)
+    else:
+        raise ValueError(f"Unrecognised data type of X, expected data type to "
+                         f"be in (np.ndarray, spmatrix, pd.DataFrame), but got :{type(X)}")
+
+    return X, precision, dtypes
+
+
+def megabytes(arr: DatasetCompressionInputType) -> float:
+    if isinstance(arr, np.ndarray):
+        memory_in_bytes = arr.nbytes
+    elif issparse(arr):
+        memory_in_bytes = arr.data.nbytes
+    elif hasattr(arr, 'iloc'):
+        memory_in_bytes = arr.memory_usage(index=True, deep=True).sum()
+    else:
+        return 0
```

Isn't it better to raise an error because 0 is obviously wrong?

---

true